### PR TITLE
Configurable warning behaviour

### PR DESCRIPTION
**Pull request by JonRowe**
An implementation for #2005

### CODE REVIEW

**Comment by JonRowe** [comment]
Anyone free to review @rspec/rspec ? Appveyor failure is due to a connectivity issue...

File: lib/rspec/core/configuration.rb [hunk 1]
```diff
@@ -1421,6 +1421,22 @@ def warnings?
...
+      # you want RSpec to run in a 'strict' no warning situation.
+      #
+      # @example
+      #
+      #   RSpec.configure do |rspec|
+      #     rspec.raise_on_warning = true
+      #   end
+      def raise_on_warning=(value)
+        if value
+          RSpec::Support.warning_notifier = lambda { |message| raise message }
```
Comment (samphippen) [inline lib/rspec/core/configuration.rb]: I think this method would be better if this lambda was also extracted to a constant, that way we'd be at the same level of abstraction on the conditional. Thoughts?

Comment (JonRowe) [inline lib/rspec/core/configuration.rb]: Don't see why not, done.

**Comment by samphippen** [comment]
LGTM. Merge on green.

**Pull request merged**